hack/generator/pkg/astmodel/identifier_factory.go

```diff
@@ -18,6 +18,8 @@ var filterRegex = regexp.MustCompile(`[\W_]`)
 type IdentifierFactory interface {
 	CreateIdentifier(name string) string
 	CreateFieldName(fieldName string) FieldName
+	CreatePrivateIdentifier(name string) string // TODO: Name?
+	CreatePrivateFieldName(fieldName string) FieldName
```

Comment:
Should these be separate methods, or should we pass an enum requesting specific visibility:

``` go
CreateFieldName(fieldName string, visibility Visibility) FieldName
```
where 
``` go
type Visibility string

const (
    Public = Visibility("public")
    Internal = Visibility("internal")
)
```